src/main/java/edu/kit/ipd/crowdcontrol/objectservice/moneytransfer/MoneyTransferManager.java

```diff
@@ -0,0 +1,172 @@
+package edu.kit.ipd.crowdcontrol.objectservice.moneytransfer;
+
+import edu.kit.ipd.crowdcontrol.objectservice.database.model.tables.records.GiftCodeRecord;
+import edu.kit.ipd.crowdcontrol.objectservice.database.model.tables.records.WorkerRecord;
+import edu.kit.ipd.crowdcontrol.objectservice.database.operations.WorkerBalanceOperations;
+import edu.kit.ipd.crowdcontrol.objectservice.database.operations.WorkerOperations;
+import edu.kit.ipd.crowdcontrol.objectservice.mail.MailHandler;
+import edu.kit.ipd.crowdcontrol.objectservice.template.Template;
+import org.jooq.Result;
+
+import java.io.*;
+import java.util.*;
+
+import javax.mail.Message;
+import javax.mail.MessagingException;
+
+/**
+ * Describes a Manager for money transfers. The Manager can log payments and pay off.
+ *
+ * @author Felix Rittler
+ */
+public class MoneyTransferManager {
+
+    MailHandler mailHandler;
+    WorkerBalanceOperations workerBalanceOperations;
+    WorkerOperations workerOperations;
+
+    int payOffThreshold;
+    int minGiftCodesCount;
+    String notificationMailAddress;
+    StringBuilder notificationText;
+
+
+    public MoneyTransferManager(MailHandler mailHandler, WorkerBalanceOperations workerBalanceOperations, WorkerOperations workerOperations, String notificationMailAddress) throws MessagingException {
+        this.mailHandler = mailHandler;
+        this.workerOperations = workerOperations;
+        this.workerBalanceOperations = workerBalanceOperations;
+        this.minGiftCodesCount = 10;
+        this.payOffThreshold = 0;
+        this.notificationMailAddress = notificationMailAddress;
+        this.notificationText = new StringBuilder();
+    }
+
+    /**
+     * Logs a new money transfer and saves it.
+     *
+     * @param workerID the id of the worker, who gets the money
+     * @param amount   the amount of money in ct
+     */
+    public void logMoneyTransfer(int workerID, int amount, int expID) {
+        workerBalanceOperations.addCredit(workerID, amount, expID);
+    }
+
+    /**
+     * Pays all workers depending on their logged money transfers.
```

Comment:
Okay looks like the entrys from above are getting payed here, maybe log is here a bit missleading, since most of the people will think of a Log like log-file where you just make a message and forget it.